python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1069,6 +1076,33 @@ def __new__(
         )
 
 
+@whitelist_for_serdes
+class ExternalAssetCheck(
```

Comment:
I think this can just directly subclass `NamedTuple` if we're not doing anything fancy in the `__new__`

Reply:
I like the runtime type checks since we're serializing it